asset/s3_exporter/processor.js

```diff
@@ -31,7 +33,14 @@ class S3Batcher extends BatchProcessor {
     }
 
     getName() {
-        const objName = `${this.objPrefix}${this.workerId}.${this.sliceCount}`;
+        let objName;
+        if (this.opConfig.compression === 'lz4') {
+            objName = `${this.objPrefix}${this.workerId}.${this.sliceCount}.lz4`;
+        } else if (this.opConfig.compression === 'gzip') {
+            objName = `${this.objPrefix}${this.workerId}.${this.sliceCount}.gz`;
+        } else {
+            objName = `${this.objPrefix}${this.workerId}.${this.sliceCount}`;
```

Comment:
If you set objName to this from the outset then you don't have to have the name expression duplicated and can get rid of the else.

Reply:
fixed